### PR TITLE
Add extended sys state for Copter

### PR DESCRIPTION
**Pull request by khancyr**
This PR add a new flag to report if we are in landing state, update the takeoff state in auto and guided mode, report land and takeoff state in the extended sys state message. Those are need for better CC control sequences !

### CODE REVIEW

**Comment by rmackay9** [comment]
This seems like a good idea in general.  I've heard a few times from GCS/companion-computer users that they had trouble knowing if the vehicle was landed or not.

**Comment by peterbarker** [comment]
Coincidentally I also implemented this on the same day that @khancyr did!

For comparison, here are the commits:
https://github.com/peterbarker/ardupilot/commit/0c205a02b4b0261c68150c8e1a00bcd4403e86ec

(just this specific chunk:)
https://github.com/peterbarker/ardupilot/commit/e5de6cea17fe3f77274f9d93d3ef7bd13f99f0f2#diff-4f15a04337259328467a92d8e35bab2fR2756

Those two functions (`vtol_state()` and `landed_state()`) are virtual-void.

**Comment by khancyr** [comment]
rebased on master, vtol_state()  (default to undefined), remove pure virtual function so all interface don't need to override it by default.

**Comment by khancyr** [comment]
rebase and update after mistake on takeoff flag (thanks @Kiwa21)

File: ArduPlane/GCS_Mavlink.h [hunk 1]
```diff
@@ -50,4 +50,5 @@ class GCS_MAVLINK_Plane : public GCS_MAVLINK
     MAV_STATE system_status() const override;
 
     uint8_t radio_in_rssi() const;
+    MAV_LANDED_STATE landed_state() const override { return MAV_LANDED_STATE_UNDEFINED;};
```
[the remark this answers is not in the document]
Comment (khancyr) [reply ArduPlane/GCS_Mavlink.h]: oups forget to remove those for now !

**Comment by khancyr** [comment]
I agree with you. But I will work on that

**Comment by WickedShell** [comment]
@khancyr Can this be closed in favor of #8587 (and if it's urgently needed then maybe we should just add ext-sys-status to a stream and put #8587 in now)

**Comment by khancyr** [comment]
Closing for  #8587

**Pull request closed without merging**